core/src/Kokkos_MathematicalFunctions.hpp

```diff
@@ -0,0 +1,232 @@
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(FUNC, double, double)           \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(FUNC, long double, long double) \
+  KOKKOS_IMPL_UNARY_FUNCTION_INTEGRAL(FUNC, double)
+
+#define KOKKOS_IMPL_MATH_UNARY_PREDICATE(FUNC)                       \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(FUNC, bool, float)       \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(FUNC, bool, double)      \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(FUNC, bool, long double) \
+  KOKKOS_IMPL_UNARY_FUNCTION_INTEGRAL(FUNC, bool)
+
+#define KOKKOS_IMPL_MATH_BINARY_FUNCTION(FUNC)                  \
+  KOKKOS_IMPL_BINARY_FUNCTION_FLOATING_POINT(FUNC, float)       \
+  KOKKOS_IMPL_BINARY_FUNCTION_FLOATING_POINT(FUNC, double)      \
+  KOKKOS_IMPL_BINARY_FUNCTION_FLOATING_POINT(FUNC, long double) \
+  KOKKOS_IMPL_BINARY_FUNCTION_ARITHMETIC(FUNC)
+
+#define KOKKOS_IMPL_MATH_NAN()                                        \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(nanf, float, char const*) \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(nan, double, char const*) \
+  KOKKOS_IMPL_UNARY_FUNCTION_FLOATING_POINT(nanl, long double, char const*)
+
+#endif
+
+// Basic operations
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(fabs)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(fmod)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(remainder)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(fmin)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(fmax)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(fdim)
+#ifndef KOKKOS_ENABLE_SYCL
+KOKKOS_IMPL_MATH_NAN()
+#endif
+// Power functions
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(pow)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(sqrt)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(cbrt)
+// Exponential functions
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(exp)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(exp2)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(expm1)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(log)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(log10)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(log2)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(log1p)
+// Trigonometric functions
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(sin)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(cos)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(tan)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(asin)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(acos)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(atan)
+KOKKOS_IMPL_MATH_BINARY_FUNCTION(atan2)
+// Hyperbolic functions
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(sinh)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(cosh)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(tanh)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(asinh)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(acosh)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(atanh)
+// Error and gamma functions
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(erf)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(erfc)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(tgamma)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(lgamma)
+// Nearest integer floating point operations
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(ceil)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(floor)
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(trunc)
+#ifndef KOKKOS_ENABLE_SYCL
+KOKKOS_IMPL_MATH_UNARY_FUNCTION(nearbyint)
```

Comment:
`nearbyint` is not provided by SYCL

Reply:
I filed an issue with Khronos to add this.  (Very unlikely to make SYCL 2020, though.)